projects/core/src/lib/users/quota/quota-info/quota-info.component.ts

```diff
@@ -0,0 +1,79 @@
+import {ChangeDetectorRef, Component, OnDestroy} from '@angular/core';
+import {ProjectService} from '../../../project/project.service';
+import {BehaviorSubject, combineLatest, mergeMap, of, Subscription} from 'rxjs';
+import {UserService} from '../../user.service';
+import {Quota} from '../quota.model';
+import {BackendService} from '../../../backend/backend.service';
+
+@Component({
+    selector: 'geoengine-quota-info',
+    templateUrl: './quota-info.component.html',
+    styleUrls: ['./quota-info.component.scss'],
+})
+export class QuotaInfoComponent implements OnDestroy {
+    sessionQuota: Quota | undefined;
+    updateQuota$: BehaviorSubject<void> = new BehaviorSubject<void>(undefined);
+    sessionSubscription: Subscription | undefined;
+    layerSubscription: Subscription | undefined;
+    plotSubscription: Subscription | undefined;
+
+    constructor(
+        protected readonly userService: UserService,
+        protected readonly projectService: ProjectService,
+        protected readonly backendService: BackendService,
+        protected readonly changeDetectorRef: ChangeDetectorRef,
+    ) {
+        this.sessionSubscription = combineLatest([
+            this.userService.isBackendFeatureEnabled('pro'),
+            this.userService.getSessionStream(),
+            this.updateQuota$,
+        ])
+            .pipe(
+                mergeMap(([isPro, _session, _update]) => {
+                    if (!isPro) {
+                        return of(undefined);
+                    }
+                    return this.userService.getSessionQuota();
+                }),
+            )
+            .subscribe((quota) => {
+                if (quota) {
+                    this.sessionQuota = Quota.fromDict(quota);
+                } else {
+                    this.sessionQuota = undefined;
+                }
+                this.changeDetectorRef.markForCheck();
+            });
+
+        this.layerSubscription = this.projectService.getLayerStream().subscribe((_layer) => {
+            this.refreshQuota();
+        });
+        this.plotSubscription = this.projectService.getPlotStream().subscribe((_plot) => {
+            this.refreshQuota();
+        });
```

Comment:
Why is this a good time to automatically update the quota?

If you want auto updates, you should think about attaching it to the same mechanism that updates the map?

Or… you could do it every 30 seconds or so.

What do you think?

Reply:
all of them are not a good trigger.

Reply:
what we would need is something that reloads if any layer or plot is loading (and when a layer is finished).

Reply:
it uses a timer now